src/main/java/com/thirdcc/webapp/service/impl/EventActivityServiceImpl.java

```diff
@@ -42,11 +52,56 @@ public EventActivityServiceImpl(EventActivityRepository eventActivityRepository,
     @Override
     public EventActivityDTO save(EventActivityDTO eventActivityDTO) {
         log.debug("Request to save EventActivity : {}", eventActivityDTO);
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.OPEN);
+            add(EventStatus.POSTPONED);
+        }};
+        Event event = eventRepository
+            .findOneByIdAndStatusIn(eventActivityDTO.getEventId(), eventStatuses)
```

Comment:
The `findOneByIdAndStatusIn` method will find the event class's status that match any of the value in eventStatuses?

Reply:
it will find the event from database with ID and status with match any value in eventStatuses

Reply:
@junzai97 Check on [JPA Repositories](https://docs.spring.io/spring-data/jpa/docs/1.5.0.RELEASE/reference/html/jpa.repositories.html) 2.3

Reply:
reference: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation

Spring Data JPA will help us to generate the sql query from the method name in `repository`

```
public interface EventRepository extends JPARepository<Event, Long> {

  Optional<Event> findOneByIdAndStatusIn(Long id, Set<Event> eventStatuses);
}
```
will be generated to sql query
```
SELECT * FROM event WHERE id = 1 AND status IN ("OPEN", "POSTPONED");
```